be/CMakeLists.txt

```diff
@@ -720,6 +727,13 @@ if ("${CMAKE_BUILD_TARGET_ARCH}" STREQUAL "x86" OR "${CMAKE_BUILD_TARGET_ARCH}"
     )
 endif()
 
+if (${WITH_LZO} STREQUAL "ON")
+    set(STARROCKS_DEPENDENCIES
+        ${STARROCKS_DEPENDENCIES}
+        lzo
+    )
+endif()
+
```

Comment:
you need to `set(CXX_COMMON_FLAGS "${CXX_COMMON_FLAGS} -DWITH_LZO") ` when ENV{WITH_LZO} is true, then the compiler will define the WITH_LZO macro when compiling the code.